cmd/thanos/compact.go

```diff
@@ -458,6 +463,41 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.compactionProgressMetrics {
+		g.Add(func() error {
+			if err := sy.SyncMetas(context.Background()); err != nil {
+				return errors.Wrapf(err, "could not sync metas")
+			}
+			originalMetas := sy.Metas()
+
+			groups, err := grouper.Groups(originalMetas)
+			if err != nil {
+				return errors.Wrapf(err, "could not group original metadata")
+			}
+
+			ps := compact.NewDefaultPlanSim(reg, tempTSDBPlanner)
+			ds := compact.NewDefaultDownsampleSim(reg)
```

Comment:
downsample one should be enabled only when downsampling is enabled in compactor.